### DISCUSSION TITLE
docker pull eclipse/zenoh-bridge-ros1 -> manifest unknown

### DISCUSSION THREAD

**Discussion by lucasw** [eclipse-zenoh/roadmap#93] [Q&A] [answered]
Following https://github.com/eclipse-zenoh/zenoh-plugin-ros1#docker-image

I'm guessing this has nothing to do with zenoh except if this is common and there is a fix some text could be added to the zenoh docker instructions.

```
$ docker pull eclipse/zenoh-bridge-ros1
Using default tag: latest
Error response from daemon: manifest for eclipse/zenoh-bridge-ros1:latest not found: manifest unknown: manifest unknown
```

Could there be something wrong with the image as setup on https://hub.docker.com/r/eclipse/zenoh-bridge-ros1?

I made a dockerhub account and did a `docker login`, but still the same result, and tried on a couple different computers (though similarly configured).  Does it work for anyone else (that doesn't already have the image pulled down)?

**Comment by lucasw** [comment]
I tried `docker pull eclipse/ditto-ui` and that starts pulling layers as expected, but eclipse/zenoh-bridge-ros2dds fails, so maybe something unique about the eclipse/zenoh docker images?

Some more:

docker pull eclipse/zenoh -> works
docker pull eclipse/platformreleng-centos-swt-build -> manifest unknown

**Reply by lucasw** [reply]
Tried again today and `eclipse/zenoh-bridge-ros2dds` still fails, but `eclipse/zenoh-bridge-ros1` is now working.

Yesterday I made a dockerfile to build the ros1 bridge from source: https://github.com/lucasw/zenoh-plugin-ros1/blob/dockerfile_apt/Dockerfile.build (I also made [one to use the apt installable versions](https://github.com/lucasw/zenoh-plugin-ros1/blob/dockerfile_apt/Dockerfile.apt) but those failed out with ```thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: PatternNotMatched', zenoh-plugin-ros1/src/ros_to_zenoh_bridge/discovery.rs:194:10`)```

**Reply by JEnoch** [reply, marked as answer]
This `manifest unknown` occurs when the `latest` tag (default tag) doesn't exist in Docker hub.

That was the case for `eclipse/zenoh-bridge-ros1 ` because its very first version wasn't released yet. But it just had been released 5 hours ago.
And this is the case now for `eclipse/zenoh-bridge-ros2dds` because that's still [work in progress](https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds), not yet released. If you want to test a nightly build of it, do `docker pull eclipse/zenoh-bridge-ros2dds:nightly`

**Reply by lucasw** [reply]
Makes sense, though it would be nice if the dockerhub main pages indicated that (and didn't have a prominent-but-invalid docker pull command)- I suppose if I used it more I would know to check the tags pages like https://hub.docker.com/r/eclipse/zenoh-bridge-ros1/tags for valid tags.
